File: send-to-server/results_to_server.py
import requests
import asyncio

url = 'http://192.168.0.75:8080/api/photogroup/photogroup_upload'

image_paths = ['../results/1.jpg', '../results/2.jpg', '../results/3.jpg', '../results/4.jpg']
video_path = '../results/1.mp4'

files = {}

for i, image_path in enumerate(image_paths):
    files['photo_' + str(i)] = open(image_path, 'rb')
files['video'] = open(video_path, 'rb')

other_data = {
    'user_id': 5,
    'season': 'spring',
    'frame':1
}

async def send_request():
    try:
        response = await requests.post(url, files=files, data=other_data)
        print(response.text)
    except Exception as e:
        print(f'에러 발생: {e}')
        
asyncio.run(send_request())

File: send_to_server/results_to_server.js
const yaml = require("js-yaml");
const fs = require('fs');
const axios = require('axios');

try {
    const yaml_file = yaml.readFileSync('config.yaml', 'utf8');
    const parsedData = yaml.safeLoad(yaml_file);
    const serverAddress = parsedData.server;
    console.info(parsedData);
} catch (e) {
    console.error(e);
}

const imagePaths = ['../results/1.jpg', '../results/2.jpg', '../results/3.jpg', '../results/4.jpg'];
const videoPath = '../results/1.mp4';

const formData = new FormData();
imagePaths.forEach((imagePath, index) => {
    const fileStream = fs.createReadStream(imagePath);
    formData.append(`image${index + 1}`, fileStream);
});
formData.append('video', fs.createReadStream(videoPath));

const otherData = {
    'user_id': 1,
    'season': 'winter',
    'frame': 1
};
Object.keys(otherData).forEach(key => {
    formData.append(key, otherData[key]);
});

axios.post(serverAddress, formData, {
    headers: {
        'Content-Type': 'multipart/form-data',
    },
})
.then(response => {
    console.log(response.data);
})
.catch(error => {
    console.error('에러 발생:', error);
});